### PR TITLE
Add some pirate gear

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

SUMMARY: Content "Add head bandana, eyepatch and baldric holster"
#### Purpose of change

Add craftable eyepatches and baldric holsters holding up to six pistols, as well as head bandanas.
#### Describe the solution

Add JSON entries for items and recipes to make them.
#### Describe alternatives you've considered

Not letting player characters dress up themselves as pirates.
#### Testing

Loaded game with copy-pasted changes, made sure that baldric holster functions correctly by storing and wielding flintlock pistols from it, made sure that no errors come up after loading because of previously-existing bandanas.
#### Additional context

Items currently don't spawn anywhere and can only be crafted; any suggestions on which item groups they could be placed into are welcome.

### CODE REVIEW

File: data/json/items/armor/holster.json [hunk 1]
```diff
@@ -30,6 +30,82 @@
     "use_action": { "type": "holster" },
     "flags": [ "BELTED", "OVERSIZE", "NO_QUICKDRAW" ]
   },
+  {
+    "id": "baldric_holster",
+    "type": "ARMOR",
+    "name": { "str": "baldric holster" },
+    "description": "A leather harness which can hold up to six pistol-sized guns in total, available for quick drawing.  Historically similar harnesses were used by pirates to carry multiple loaded flintlock pistols at once, as reloading them is a lenghty process.  Activate to holster/draw a gun.",
```
Comment (LaVeyanFiend) [inline data/json/items/armor/holster.json:37]:
```suggestion
    "description": "A leather harness which can hold up to six pistol-sized guns in total, available for quick drawing.  Historically similar harnesses were used by pirates to carry multiple loaded flintlock pistols at once, as reloading them is a lengthy process.  Activate to holster/draw a gun.",
```

**Comment by Theawesomeboophis** [comment]
Ah yes, this will do nicely on my modded medieval polish warrior with squid legs and a toothbrush moustache.
Just another normal day in cataclysm.

**Comment by curstwist** [comment]
Please put the eye patch and scarf into the item_group: `costume_clothes` to fill out that group some more.

**Comment by dissociativity** [comment]
can you make a bandana 'a'ctivate into a head bandana and vice versa? in real life they're rather versatile so it's silly to restrict them to the mouth

**Comment by Rail-Runner** [comment]
> Please put the eye patch and scarf into the item_group: `costume_clothes` to fill out that group some more.
> 
> edit: make that `costume_accessories`

Done.

> can you make a bandana 'a'ctivate into a head bandana and vice versa? in real life they're rather versatile so it's silly to restrict them to the mouth

That's what I did already; I added use actions to both bandana items.

File: data/json/items/armor/masks.json [hunk 2]
```diff
@@ -30,14 +30,14 @@
...
-    "type": "ARMOR",
+    "type": "TOOL_ARMOR",
     "name": { "str": "bandana" },
     "description": "A cotton bandana, worn over the mouth for warmth and minor protection from dust and other contaminants.",
     "weight": "42 g",
     "volume": "250 ml",
     "price": 600,
     "price_postapoc": 10,
-    "to_hit": -1,
+    "use_action": { "type": "transform", "msg": "You adjust your %s to cover head.", "target": "bandana_head", "menu_text": "Cover head" },
```
Comment (chaohedgehogs) [inline data/json/items/armor/masks.json:40]:
```suggestion
    "use_action": { "type": "transform", "msg": "You adjust your %s to cover your head.", "target": "bandana_head", "menu_text": "Cover head" },
```

File: data/json/items/armor/hats.json [hunk 3]
```diff
@@ -21,6 +21,25 @@
...
+  {
+    "id": "bandana_head",
+    "type": "TOOL_ARMOR",
+    "name": { "str": "head bandana" },
+    "description": "A cotton bandana, worn over the head.",
+    "weight": "42 g",
+    "volume": "250 ml",
+    "price": 600,
+    "price_postapoc": 10,
+    "use_action": { "type": "transform", "msg": "You adjust your %s to cover mouth.", "target": "bandana", "menu_text": "Cover mouth" },
```
Comment (chaohedgehogs) [inline data/json/items/armor/hats.json:33]:
```suggestion
    "use_action": { "type": "transform", "msg": "You adjust your %s to cover your mouth.", "target": "bandana", "menu_text": "Cover mouth" },
```

**Pull request merged**